packages/contract/src/AccountFactory.sol

```diff
@@ -26,19 +28,25 @@ contract AccountFactory {
      * This method returns an existing account address so that entryPoint.getSenderAddress() would work even after account creation
      */
     function createAccount(
-        bytes32[2] memory accountKey,
+        uint8 slot,
+        bytes32[2] memory key,
+        Call[] calldata initCalls,
         uint256 salt
-    ) public returns (Account ret) {
-        address addr = getAddress(accountKey, salt);
+    ) public payable returns (Account ret) {
+        address addr = getAddress(slot, key, initCalls, salt);
         uint codeSize = addr.code.length;
         if (codeSize > 0) {
             return Account(payable(addr));
         }
+
+        // Prefund the account with msg.value
+        entryPoint.depositTo{value: msg.value}(addr);
```

Comment:
might wrap this in `if (msg.value > 0)`

saves us a contract call when using paymaster

Reply:
done